client/my-sites/customer-home/cards/features/stats-v2/index.jsx

```diff
@@ -0,0 +1,265 @@
+							</div>
+							{ showTopPost && (
+								<div className="stats-v2__data-item">
+									<div className="stats-v2__data-label">{ translate( 'Top post' ) }</div>
+									<div className="stats-v2__data-value">{ topPost.title }</div>
+								</div>
+							) }
+							{ showTopPage && (
+								<div className="stats-v2__data-item">
+									<div className="stats-v2__data-label">{ translate( 'Top page' ) }</div>
+									<div className="stats-v2__data-value">{ topPage.title }</div>
+								</div>
+							) }
+							{ showViews && (
+								<div className="stats-v2__data-item">
+									<div className="stats-v2__data-label">{ translate( 'Total views' ) }</div>
+									<div className="stats-v2__data-value">{ numberFormat( views ) }</div>
+								</div>
+							) }
+							{ showVisitors && (
+								<div className="stats-v2__data-item">
+									<div className="stats-v2__data-label">{ translate( 'Total visitors' ) }</div>
+									<div className="stats-v2__data-value">{ numberFormat( visitors ) }</div>
+								</div>
+							) }
+						</div>
+						<a href={ `/stats/day/${ siteSlug }` } className="stats-v2__all">
+							{ translate( 'See all stats' ) }
+						</a>
+					</>
+				) }
+			</Card>
+		</div>
+	);
+};
+
+const mapStateToProps = ( state ) => {
+	const siteId = getSelectedSiteId( state );
+	const siteSlug = getSelectedSiteSlug( state );
+	const isSiteUnlaunched = isUnlaunchedSite( state, siteId );
+	const gmtOffset = getSiteOption( state, siteId, 'gmt_offset' );
+	const date = moment()
+		.utcOffset( Number.isFinite( gmtOffset ) ? gmtOffset : 0 )
+		.format( 'YYYY-MM-DD' );
+	const period = 'day';
+	const quantity = 7;
+
+	const chartTab = 'views';
+	const chartQuery = {
+		chartTab,
+		date,
+		period,
+		quantity,
+		siteId,
+		statFields: [ chartTab ],
+	};
+	let chartData;
+
+	let topPost;
+	let topPage;
+	const topPostsQuery = {
+		date,
+		num: quantity,
+		period,
+	};
+
+	let views;
+	let visitors;
+	const visitsQuery = {
+		unit: period,
+		quantity: quantity,
+		stat_fields: 'views,visitors',
+	};
+
+	let mostPopularDay;
+	let mostPopularTime;
+	const insightsQuery = {};
+
+	const isLoading = isLoadingStats(
+		state,
+		siteId,
+		chartTab,
+		period,
+		insightsQuery,
+		topPostsQuery,
+		visitsQuery
+	);
+
+	if ( ! isSiteUnlaunched && ! isLoading ) {
+		const counts = getCountRecords( state, siteId, period );
+		chartData = buildChartData( [], chartTab, counts, period, date );
+
+		const mostPopularDateTime = getMostPopularDatetime( state, siteId, insightsQuery );
+		mostPopularDay = mostPopularDateTime.day ?? '-';
+		mostPopularTime = mostPopularDateTime?.time ?? '-';
+
+		const topPostAndPage = getTopPostAndPage( state, siteId, topPostsQuery );
+		topPost = topPostAndPage.post;
+		topPage = topPostAndPage.page;
+
+		const viewsAndVisitors = getViewAndVisitors( state, siteId, visitsQuery );
+		views = viewsAndVisitors.views;
+		visitors = viewsAndVisitors.visitors;
+	}
```

Comment:
Non-blocking, but the function here might be long enough to add a few helper functions to improve readability.

Reply:
https://github.com/Automattic/wp-calypso/pull/41441/commits/1766e698b7e52737ef2beb36d8ce7293a976b014